internal/testharness/pty/appfixture/config.go

```
@@ -0,0 +1,43 @@
+	config := fmt.Sprintf("model = \"gpt-5\"\nprovider_override = \"openai\"\nopenai_base_url = \"http://127.0.0.1:1/v1\"\nserver_port = %d\n\n[reviewer]\nfrequency = \"off\"\n", port)
+	if err := os.WriteFile(configPath, []byte(config), 0o644); err != nil {
+		return fmt.Errorf("write fixture config: %w", err)
+	}
+	if _, err := metadata.RegisterBinding(ctx, persistenceRoot, workspaceRoot); err != nil {
+		return fmt.Errorf("register fixture workspace binding: %w", err)
+	}
+	return nil
+}
+
+func freeTCPPort() (int, error) {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)